src/Synchronizer/VariantConfigurationSynchronizer.php

```diff
@@ -22,30 +26,40 @@ final class VariantConfigurationSynchronizer implements VariantConfigurationSync
 
     private VariantCollectionInterface $variantCollection;
 
+    private ClientInterface $client;
+
     public function __construct(
         FactoryInterface $variantConfigurationFactory,
         VariantConfigurationRepositoryInterface $variantConfigurationRepository,
         VariantCollectionInterface $variantCollection,
-        ManagerRegistry $managerRegistry
+        ManagerRegistry $managerRegistry,
+        ClientInterface $client
     ) {
         $this->variantConfigurationFactory = $variantConfigurationFactory;
         $this->variantConfigurationRepository = $variantConfigurationRepository;
         $this->variantCollection = $variantCollection;
         $this->managerRegistry = $managerRegistry;
+        $this->client = $client;
     }
 
-    public function synchronize(): void
+    public function synchronize(bool $createVariantsIfNotExists): VariantCollectionInterface
```

Comment:
This couples code that shouldn't be coupled. I.e. in this case it couples the syncing of variant configuration with the creation of variants in Cloudflare. Instead this synchronizer should dispatch an event you can listen to and act on accordingly

Reply:
Well, I thought the whole idea of https://github.com/Setono/SyliusImagePlugin/blob/master/src/Config/Variant.php#L98 was to ensure that they can be created by whatever generator was chosen. Otherwise, why the requirement for certain properties?

Reply:
Creating 'instructions' for variants (i.e. on Cloudflare) is not the same as actually generating the images (which is what the `VariantGeneratorInterface` does)

Reply:
Okay, sure. But we're still imposing requirements on configuration (requiring size params) for something that is only Cloudflare specific? If we impose those requirements, I believe the utilization of them should be generalized.

We could add an interface that specifies functions to create and update variants, and let it be optional if the generator chooses to implement it. When handling a variant, just check whether the generator found in the registry implements that interface. If it doesn't, it's okay, we just cant provide automatic variant creation/update. If it does, fine - we can manage the variants then.

Reply:
I think if we really should make this generic, we should have width and height nullable (but not both at the same time) and require a fit.

Reply:
Well, seems you were on the right track back then, then ;)

No, but I really feel that right now, the implementation sits "between two chairs" imposing requirements on configuration and at the same time not generalizing the utilization of the required values.

With a generalization of variant creation/update it could be brought to the surface area (i.e. allowing this argument to the `synchronize` method), improving the visibility/discoverability/"configurability" of the functionality, where as right now it's "hidden" in an event subscriber with no easy way to disable it, except for doing a compiler pass and removing the subscriber.